### PR TITLE
add translation for System Design, Scalability, Data Handling

### PR DESCRIPTION
**Pull request by thovo**
@letientai299 : Please review!

### CODE REVIEW

File: README-vn.md [hunk 1]
```diff
@@ -976,4 +976,130 @@ Bạn sẽ biết thêm nhiều ứng dụng của đồ thị trong sách của
 
 > TODO Add anchors link for the "danh mục sách"
 
+---
+
+## Thiết kế hệ thống, Khả năng mở rộng, Xử lý dữ liệu
+- **Bạn có thế sẽ bị hỏi câu hỏi liên quan đến thiết kế hệ thống nếu có hơn 4 năm kinh nghiệm**
+- Khả năng mở rộng và Thiết kế hệ thống là một chủ đề rất rộng với nhiều chủ đề và tài liệu, do có rất nhiều vấn đề cần phải giải quyết khi thiết kế phần mềm hoặc phần cứng có thể mở rộng được.
```
Comment (letientai299) [inline README-vn.md]: > Khả năng mở rộng và Thiết kế hệ thống là **một** chủ đề rất rộng

Should be "hai" or "những".

> do có rất nhiều vấn đề cần phải giải quyết khi thiết kế phần mềm hoặc phần cứng có thể mở rộng được.

Kind of confusing.

File: README-vn.md [hunk 2]
```diff
@@ -976,4 +976,130 @@ Bạn sẽ biết thêm nhiều ứng dụng của đồ thị trong sách của
...
+    - [ ] Twitter:
+        - [O'Reilly MySQL CE 2011: Jeremy Cole, "Big and Small Data at @Twitter" (video)](https://www.youtube.com/watch?v=5cKTP36HVgI)
+        - [Timelines at Scale](https://www.infoq.com/presentations/Twitter-Timeline-Scalability)
+    - Để có thêm thông tin, xem seri "Mining Massive Datasets" trong seri video.
+- [ ] Thực hành thêm về thiết kế hệ thống: Sau đây là vài ý tưởng có thể thực hiện trên giấy, mỗi ý tưởng đều có tư liệu đi cùng để hiểu rõ nó thực thi thế nào trong thế giới thực. [System Design from HiredInTech - Thiết kế hệ thống bới HiredInTech](http://www.hiredintech.com/system-design/)
+    - [cheat sheet](https://github.com/jwasham/google-interview-university/blob/master/extras/cheat%20sheets/system-design.pdf)
+    - Các bước thực hiện:
+        1. Hiểu vấn đề và phạm vi của vấn đề:
+            - Định nghĩa các trường hợp sử dụng cụ thể với sự trợ giúp từ người phỏng vấn
+            - Đề nghị thêm tính năng
```
Comment (letientai299) [inline README-vn.md]: > Đề nghị
 I think "Đề xuất" is better for this case.

File: README-vn.md [hunk 3]
```diff
@@ -976,4 +976,130 @@ Bạn sẽ biết thêm nhiều ứng dụng của đồ thị trong sách của
...
+        2. Suy nghĩ về các ràng buộc:
+            - Hỏi xem có bao nhiêu yêu cầu mỗi tháng
+            - Hỏi xem có bao nhiêu yêu cầu mỗi giây (hoặc họ đưa nó cho bạn hoặc bạn phải tự tính toán ra)
+            - So sánh tần số đọc và ghi
+            - Sử dụng luật 80/20 khi ước lượng
+            - Bao nhiêu dữ liệu được viết mỗi giây
+            - Toàn bộ kho lưu trữ dữ liệu cần cho 5 năm
+            - Bao nhiêu dữ liệu được đọc mỗi giây
+        3. Thiết kế trừu tượng:
+            - Layers (service, data, caching)
```
Comment (letientai299) [inline README-vn.md]: Missing translation?

**Comment by letientai299** [comment]
Beside above review comment, there's one thing I suggest you to update to maintain consistency. 

For external link, usually, I will put Vietnamse translation first, then the original English come later, between the parentheses, if needed. Like this:

```
CPU thực thi chương trình thế nào (How does CPU execute program) - video
 ```

I'm sorry that I didn't review previous PRs carefully, just too lazy. But from now, please use that formmatting style. So that later, we don't have much to rework.

Thanks.

File: README-vn.md (hunk 3 above)
Comment (thovo) [reply README-vn.md]: I need some advice for this part, how to translate `layer` correct :joy:

**Comment by thovo** [comment]
That is okay, we can make it like a convention.

File: README-vn.md (hunk 1 above)
Comment (thovo) [reply README-vn.md]: Which sentence?

File: README-vn.md (hunk 2 above)
Comment (thovo) [reply README-vn.md]: Ok!

**Comment by thovo** [comment]
Please review again @letientai299 :smile: !

**Comment by letientai299** [comment]
Well done. :+1: 

There's some small issues, but I will merge anyway and fix them from my side. Thank you.

**Pull request merged**